hil/client/base.py

```diff
@@ -60,9 +62,51 @@ def check_response(self, response):
             except ValueError:  # No JSON request body; typical
                                 # For methods PUT, POST, DELETE
                 return
-        else:
+        try:
             e = json.loads(response.content)
             raise FailedAPICallException(
                 error_type=e['type'],
                 message=e['msg'],
             )
+        # Catching responses that do not return JSON
+        except ValueError:
+            return response.content
+
+
+def _find_reserved(string, slashes_ok=False):
+    """Returns a list of illegal characters in a string"""
+    if slashes_ok:
+        p = r"[^A-Za-z0-9 /$_.+!*'(),-]+"
+    else:
+        p = r"[^A-Za-z0-9 $_.+!*'(),-]+"
+    return list(x for l in re.findall(p, string) for x in l)
+
+
+def check_reserved(obj_type, obj_string, slashes_ok=False):
+    """Check for illegal characters and report of their existence"""
+    bad_chars = _find_reserved(obj_string, slashes_ok)
+    if bool(bad_chars):
+        error = obj_type + " may not contain: " + str(bad_chars)
+        raise BadArgumentError(error)
+
+
+def check_reserved_chars(*outer_args, **outer_kwargs):
+    """Wraps Client lib functions to check for illegal characters
+    and dynamically report the error by the offending argument(s)"""
+    def wrapper(f):
+        """Auxiliary wrapper for check_reserved_chars"""
+
+        def reserved_wrap(*args, **kwargs):
+            """Wrapper that is passed the arguments of the wrapped function"""
+            if 'slashes_ok' in outer_kwargs:
+                slashes_ok = outer_kwargs.get('slashes_ok')
+            else:
+                slashes_ok = []
+            for argname, argval in zip(outer_args, args[1:]):
```

Comment:
Took me a second to work out why we were chopping off the first element; probably should add a comment about that.